src/Driver/PDO/Connection.php

```diff
@@ -80,6 +80,9 @@ public function query(string $sql): ResultInterface
      */
     public function quote($value, $type = ParameterType::STRING)
     {
+        /**
+         * @var PDO::PARAM_* $type
+         */
```

Comment:
Shouldn't we rather fix the type of `$type` in `Connection::quote()`?

Reply:
I have a better solution, see the latest code

Reply:
I'm not a good solution to fix the type of `$type` in `Doctrine\DBAL\Driver\Connection::quote()`.

My solution is https://github.com/doctrine/dbal/pull/5835/commits/7b8ea5e5adf2db02c649384db0f5de525371e3a7. Maybe have a better solution. Please give any feedback.